src/lib/sql/sql_identifier_resolver_proxy.hpp

```diff
@@ -0,0 +1,42 @@
+#pragma once
+
+#include <memory>
+#include <vector>
+
+#include "expression/parameter_expression.hpp"
+
+namespace opossum {
+
+class AbstractExpression;
+struct SQLIdentifier;
+class SQLIdentifierResolver;
+class ParameterIDAllocator;
+
+/**
+ * Used during SQLTranslation to resolve "outer expressions" in sub-Selects.
+ * Consider "SELECT (SELECT t1.a + b FROM t2) FROM t2". The SQLTranslator uses the SQLIdentifierResolverProxy to resolve
+ * "t1.a" while translating the sub-Select.
```

Comment:
Could you talk a bit about the difference between the resolver and the proxy?

Reply:
Not really sure what you mean, but I extended the comment a bit. 

(The resolver is a thing. The proxy wraps that thing and does something with the accesses to it.)

Reply:
I think this is the important part: 

```
 * The SQLIdentifierResolverProxy provides and tracks access to a SELECT statement's identifiers from ANY inner
 * query, no matter how deeply nested.
```